batchspawner/batchspawner.py

```diff
@@ -775,4 +777,106 @@ def state_gethost(self):
         self.log.error("Spawner unable to match host addr in job {0} with status {1}".format(self.job_id, self.job_status))
         return
 
+
+class OARSpawner(UserEnvMixin, BatchSpawnerBase):
+    batch_script = Unicode("""#!/bin/sh
+#OAR -n jupyterhub-singleuser-{{username}}
+#OAR -l nodes=1{% if nprocs     %}/core={{nprocs}}{% endif %}{% if runtime    %},walltime={{runtime}}{% endif %}
+{% if queue      %}#OAR -q {{queue}}{% endif %}
+{% if memory     %}#OAR -p "memnode>={{memory}}"{% endif %}
+{% if options    %}#OAR {{options}}{% endif %}
+
+{{prologue}}
+{{cmd}}
+{{epilogue}}
+""").tag(config=True)
+
+    # use a script to submit to batch queueing system instead of piping it in
+    script_mode = stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH #755 by default
```

Comment:
Is there a reason for this to be 755?  The script includes environment variables in it, including the JH API tokens, so should be protected.  If it's in an admin's protected $HOME it's OK, but if it's in /tmp it could be an issue.

Reply:
This _IS_ an issue. It has to be in admin's directory and must be readable and executable by the user. I did not manage to get it running in any way not creating any security hole. 
The less-worse solution would be "burn after reading" that once the hand is given to the user destroys those sensitive information from the script.

Reply:
But if this is the best we can do, so be it.  As long as we document it properly.

If the submit script itself has to be public and has private data, could one write the sensitive data to a separate file in the user's home dir, then source that file from this script?  (Maybe this is too much if the above already meets your security requirements...)